Assets/Scripts/Models/Area/RoomBehavior.cs

```diff
@@ -426,14 +425,13 @@ private bool DefaultIsValidRoom(Room room)
                 return false;
             }
 
-            List<Tile> innerTiles = room.GetInnerTiles();
-            List<Tile> borderTiles = room.GetBorderingTiles();
-
-            List<Tile> allTiles = innerTiles.Union(borderTiles).ToList();
+            HashSet<Tile> innerTiles = room.GetInnerTiles();
+            HashSet<Tile> borderTiles = room.GetBoundaryTiles();
+            innerTiles.UnionWith(borderTiles);
```

Comment:
Same concern here as previous regarding name of innerTiles, and possibility of not creating an object for borderTiles if it isn't used.